src/edge/models/genome.py

```diff
@@ -180,6 +180,71 @@ def changed_locations_by_fragment(self):
         changes = {Fragment.objects.get(pk=f): v for f, v in changes.items()}
         return changes
 
+    def get_coordinate_diff_from_parent_genome(self, p_genome_id):
+        # Get fragments contained by child and not by parent
+        p_fragment_ids = Fragment.objects.filter(genome=p_genome_id).values_list('id', flat=True)
+        child_only_fragments = self.fragments.filter().exclude(genome=p_genome_id)
+
+        # Relate child fragments to parent fragments
+        child_to_parent_f_ids = {}
+        for f in child_only_fragments:
+            curr_f = f
+            while curr_f is not None and curr_f.id not in p_fragment_ids:
+                curr_f = curr_f.parent
+            if curr_f is not None:
+                child_to_parent_f_ids[f.id] = curr_f.id
+        if len(child_to_parent_f_ids) == 0:
+            raise Exception("Genome input is not a parent by lineage")
+
+        # Get indexed child and parent fragments
+        child_to_parent_fs = {
+            Fragment.objects.get(pk=child_f_id).indexed_fragment():
+                Fragment.objects.get(pk=parent_f_id).indexed_fragment()
```

Comment:
Would we be able to save a fetch or two, if we just use the objects fetched once at the start of this code?

Reply:
Good point, I'll grab these from the querysets instead

Reply:
Just updated this, went ahead and just fetched the objects when we generate the parent/child mapping to skip the redundancy